src/types/redis_json.cc

```diff
@@ -573,6 +575,9 @@ rocksdb::Status Json::MSet(engine::Context &ctx, const std::vector<std::string>
 
   auto batch = storage_->GetWriteBatchBase();
   WriteBatchLogData log_data(kRedisJson);
+
+  std::unordered_map<std::string, std::pair<JsonValue, JsonMetadata>> dirty_keys{};
```

Comment:
```suggestion
  // A single JSON key may be modified multiple times in the MSET command,
  // so we need to record them temporarily to avoid reading old values from DB.
  std::unordered_map<std::string, std::pair<JsonValue, JsonMetadata>> dirty_keys{};
```